### PR TITLE
fix: repository page speed improvements

### LINKED ISSUE #3278: bug: repository page times out frequently
The repository page is crashing frequently with lambda timeout errors.

We should return minimal info server side. Currently there is some data that can take a longer time to be calculated. This should be handle on the client-side.

Relates to https://github.com/open-sauced/api/issues/750

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

Brings repository page speed improvements to decrease the number of lambda timeouts we've been seeing.

## Related Tickets & Documents

Fixes #3278

Relates/Depends on https://github.com/open-sauced/api/issues/750#issuecomment-2079488228

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [x] Tier 4

### CODE REVIEW

**Comment by nickytonline** [comment]
This is ready to go, but I'm going to leave it in draft until https://github.com/open-sauced/api/pulls/750 is merged as this PR depends on it.

**Comment by nickytonline** [comment]
The tests are failing because of the lamdba timeouts in repo pages that this PR fixes. ♻️

File: lib/hooks/api/useRepoStats.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import useSWR, { Fetcher } from "swr";
```
Comment (nickytonline) [inline lib/hooks/api/useRepoStats.ts:1]: A new hook to retrieve repository stats. Prwviously, this data was being loaded in `getServerSideProps` which was causing frequent timeouts.

**Review by brandonroberts — APPROVED** [review]
⚡

**Review by zeucapua — APPROVED** [review]

**Comment by nickytonline** [comment]
Here's the real comparison now that the fix is deployed to production.

**Before**

image

**After**

image

**Comment by nickytonline** [comment]
Relates to this OKR, https://github.com/open-sauced/engineering/issues/292

**Pull request merged**